apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/Emulator/metadata_server.py

```diff
@@ -18,13 +18,135 @@
 from werkzeug.wrappers import Request
 import common, errno, getopt, json, logging, os, socket, string, sys, time
 
-numeric_project_id = 12345
-numeric_project_id_token = u'<NUMERIC_PROJECT_ID>'
-project_id = u'fake-project'
-project_id_token = u'<PROJECT_ID>'
+class ServerState:
+  fake_numeric_project_id = 12345
+  fake_project_id = u'fake-project'
+
+  _credential_scopes = ["https://www.googleapis.com/auth/cloud-platform",
+                        "https://www.googleapis.com/auth/userinfo.email"]
+
+  def __init__(self, is_for_testing, project_id, numeric_project_id, use_default_credentials, credentials_files):
+    if project_id:
+      self.project_id = project_id
+    elif is_for_testing:
+      self.project_id = ServerState.fake_project_id
+    else:
+      self.project_id = os.environ.get('TEST_PROJECT', ServerState.fake_project_id)
+
+    if numeric_project_id:
+      self.numeric_project_id = numeric_project_id
+    else:
+      self.numeric_project_id = ServerState.fake_numeric_project_id
+
+    self._credentials = {}
+    if use_default_credentials:
+      self._useDefaultCredentials()
+    if credentials_files:
+      self._useCredentials(credentials_files)
```

Comment:
if both use_default_credentials are and credentials_files are set, what do you want to happen? since you are mutating the map, you're going to potentially get entries from both... is that desired?